### ISSUE TITLE
API for returning transactions

### ISSUE THREAD

**Issue by tschubotz** [gnosis/safe-relay-service#105]
# What is needed?
We need  GET /transactions endpoint to return failed, pending and incoming transactions.

# TODO
- [ ] define what Safes are tracked
- [ ] define request
- [ ] define response
- [ ] what about pending transactions not submitted via the relay

**Comment by tschubotz** [comment]
Notes from service sync:
_Uxio, Denis, Richard, Lukas, Tobi_

What are "pending txs"?
- Could be just the txs submitted via our service
- Could be txs submitted to the blockchain by anyone (i.e. not only submitted via our service)
- For the team Safe, there is a difference between pending (on the chain) and awaiting (i.e. waiting for confirmations)

Tracing is currently only done on mined txs, *not* pending txs.
-> Similar to when you e.g. submit a txs via Metamask, but Etherscan takes a while to pick it up.

Would we see pending tx submitted via our service?
- For mobile apps, service has tx in database, so generally yes.
- For team Safe, txs are not submitted by the service, but rather via the used wallet, i.e. the service does not know about it i.e. only the executing/submitting user would see it as pending. The others would see it upon successful / failed execution.
-> Let's improve this in a future iteration for the team Safe.

tldr; 
Endpoint first for mined transactions. Later on, improve to include pending txs not submitted via the service.

@Uxio0 to create a ticket for some kind of big refactoring and team version 0.2.0 ;)

**Comment by Uxio0** [comment]
Here it's a first draft Richard and I discussed yesterday:

### Notes
- Avoid polimorphic lists
- Always send the default transaction information, that will allow clients that cannot handle a specific type to just display generic information
    - We can add new types without breaking old clients
- Should be paginated

### Open questions
- What happens to multi send (or similar)
    - multiple Safe tx/ transfers in one Ethereum tx

### Proposed structure
- object
    - <pagination info> ...
	- transactions: list
		- id: <string> // to link to the details
		- hash: <hex string> // ethereum transaction hash
		- from: <eth address>
		- to: <eth address> // TBD: nullable for create?
		- value: <stringyfied-int> // TBD: alternatively a hex?
		- data: <hex string> // TBD: nullable?
		- mined: <unix timestamp> // this should be a int not a string
		- status: <boolean>
        - safeTx: // Only for outgoing Safe Txs. I would return everything so clients will be able to check the safeTxHash
            - safe: <eth address>
            - safeTxHash: <hex string>
            - to: <eth address>
            - value: <stringyfied-int>
            - data: <hex string> // TBD: nullable?
            - operation: <enum>
            - safeTxGas: <stringyfied-int>
            - baseGas: <stringyfied-int>
            - gasPrice: <stringyfied-int>
            - gasToken: <eth address>
            - safeNonce: <stringyfied-int>
            - refundReceiver: <eth address>
            - decoded: // Object with parameters decoded if call to the safe contract
                - methodName: <string> // e.g. _approveHash_
                - parameters:
                    - whatever // e.g. _hashToApprove: '0xabc...dc'_ 
            - submitted: <unix timestamp> //  nullable -> for incoming; this should be a int not a string
	    - erc20: list // empty list
			- token: <eth address>
			- from: <eth address>
			- to: <eth address>
			- value: <stringyfied-int> // TBD: alternatively a hex?
	    - erc721: list // empty list
			- token: <eth address>
			- tokenID: <stringyfied-int> // TBD: alternatively a hex?
			- from: <eth address>
			- to: <eth address>

	    - <potential other type>: object // nullable
		    -<id - <string> >: object
			    - ...

**Comment by rmeissner** [comment]
There are still some open questions. Please check the document and comment either here or directly on the document

**Comment by tschubotz** [comment]
@Uxio0 would like to discuss this with Richard and the rest of the team. 
Uxio and @rmeissner will set up a meeting with the relevant people once Richard is back :)

**Comment by rmeissner** [comment]
This issue should be moved to the transaction service once #122 is done
